src/stylus/CargoStylusFactory.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+interface IArbWasm {
+    function activateProgram(
+        address program
+    ) external payable returns (uint16 version, uint256 dataFee);
+}
+
+contract CargoStylusFactory {
+    // Constants
+    address constant ARB_WASM = 0x0000000000000000000000000000000000000071;
+
+    // Events
+    event ContractDeployed(address indexed deployedContract, address indexed deployer);
+    event ContractActivated(address indexed activatedContract);
+    event ContractInitialized(address indexed initializedContract);
+
+    // Errors
+    error ContractDeploymentError(bytes bytecode);
+    error ContractInitializationError(address newContract);
+    error RefundExcessValueError(uint256 excessValue);
+    error ExecutionFailed(address target, bytes data);
+
+    // Deploys, activate and inits a new contract
+    function deployActivateInit(
+        bytes calldata _bytecode,
+        bytes calldata _constructorCalldata,
+        uint256 _constructorValue
+    ) public payable returns (address) {
+        uint256 activationValue = msg.value - _constructorValue;
+
+        // Deploy the contract
+        address newContractAddress = deployContract(_bytecode);
+
+        // Activate the contract
+        uint256 activationFee = activateContract(newContractAddress, activationValue);
+
+        // Initialize the contract
+        initializeContract(newContractAddress, _constructorCalldata, _constructorValue);
+
+        refundExcessValue(activationFee, _constructorValue);
```

Comment:
this ignore any refund from the constructor call (if any), I think we can wrap a balance check before and after the call to constructor to know the net amount spent in the constructor and do refund appropriately

there is also an additional edge case where the constructor cost is negative (if it somehow received more fund from the constructor)